scipy/special/_basic.py

```diff
@@ -2639,22 +2637,20 @@ def comb(N, k, exact=False, repetition=False, legacy=True):
     220
 
     """
+    if legacy:
+        warnings.warn(
+            "Using 'legacy=True' is deprecated and will raise an error by "
+            "Scipy 1.13.0. If you want to keep the legacy behaviour, cast "
+            "your inputs directly, e.g. "
+            "'comb(int(your_N), int(your_k), exact=True)'.",
+            DeprecationWarning,
+            stacklevel=2
+        )
     if repetition:
         return comb(N + k - 1, k, exact, legacy=legacy)
     if exact:
         if int(N) != N or int(k) != k:
```

Comment:
This should now be redone; the conditions were only there to make the warning less noisy
```suggestion
    if exact:
        if int(N) == N and int(k) == k:
            # _comb_int casts inputs to integers, which is safe & intended here
            return _comb_int(N, k)
        elif legacy:
            # here at least one number is not an integer; legacy behavior just casts to int
            return _comb_int(N, k)
        # otherwise, we disregard `exact=True`; it makes no sense for non-integral arguments
        return comb(N, k)
    else:
```